src/torrent.cpp

```diff
@@ -8288,8 +8260,16 @@ namespace libtorrent
 
 		int ret = 0;
 
-		std::int64_t const fin_time = finished_time();
-		std::int64_t const download_time = int(active_time()) - fin_time;
+		const seconds32 act_time = active_time();
+		static_assert(std::is_same<
+			decltype(seed_rank_priority_duration), decltype(act_time)>::value,
+			"Need to be same ratio type to prevent runtime conversion");
+
+		// use raw duration values because active time and finish time use same
+		// ratio type and don't need to be converted to a equal ratio
+		// and we don't compare them agains seconds we just use fraction value
+		std::int32_t const fin_time = finished_time().count();
+		std::int32_t const download_time = act_time.count() - fin_time;
```

Comment:
I don't want to divide inside of sort code, because it's not neccesary if we only need a ration between 2 values of equal type.
Idea is not too drop the performance. If I implement duration here it would maybe slower then before.

If you don't care about it I could just use durations instead of count() + compile time typecheck.